docs/guides/migration/10.0.0.md

```diff
@@ -0,0 +1,108 @@
+# Migration to 10.0.0
+This guide describes all breaking changes introduced with `v10.0.0`.
+
+### `pollForQueryResponse` returns response as string ([#1285](https://github.com/aeternity/aepp-sdk-js/pull/1285))
+
+- replace `oldResult.decode` with `new Buffer(newResult)`.
+- replace `oldResult.response` with `TxBuilderHelper.encode(new Buffer(newResult), 'or'')`.
+
+### removed `skipArgsConvert` option of contract call and deployment ([6d4a599](https://github.com/aeternity/aepp-sdk-js/commit/6d4a59986d6833866b6828085fad96b364e5d315))
+
+Convert arguments in intermediate Sophia representation to JavaScript types.
+For example:
+```js
+contract.methods.listFn('[1, 2]', { skipArgsConvert: false })
+```
+rewrite to
+```js
+contract.methods.listFn([1, 2])
+```
+
+### removed `skipTransformDecoded` option of contract call and deployment ([bb49239](https://github.com/aeternity/aepp-sdk-js/commit/bb492396a29d673f02eee7cfec4e114c2fba2f3c))
+
+Decoding to JavaScript types is enforced, please use it instead.
+
+### ak_ addresses are not accepted as hashes, bytes, and signatures anymore ([cbaac62](https://github.com/aeternity/aepp-sdk-js/commit/cbaac6263dd1729d64ef3a01c94e10687fed3b0d))
+
+Encode addresses as an `ak_`-prefixed string instead.
+
+### removed `contractEncodeCall` ([a4b303f](https://github.com/aeternity/aepp-sdk-js/commit/a4b303fbcc3b7cc544a2b5b3415d0f4a147b488c))
+
+Use `contractEncodeCallDataAPI` instead.
+
+### removed `contractDecodeData` ([5df2285](https://github.com/aeternity/aepp-sdk-js/commit/5df2285a7b5694475554e5154c544e0885bd4b33))
+
+Use `contractDecodeCallResultAPI` instead.
+
+### removed `setOptions` on contract instance ([b88e767](https://github.com/aeternity/aepp-sdk-js/commit/b88e767c86874f259ff7a1eb1784c368524e7167))
+
+Pass them through `getContractInstance` options instead.
+
+### additional options of `getContractInstance` accepted as usual ones ([10fb7ba](https://github.com/aeternity/aepp-sdk-js/commit/10fb7bad8f61b973c1be5daec50909d251ec1f90))
+
+For example, replace
+```js
+sdk.getContractInstance(source, { opt: { ttl: 1 } })
+```
+with
+```js
+sdk.getContractInstance(source, { ttl: 1 })
```

Comment:
is that still correct with the source as part of the object changes?

Reply:
I haven't checked the guide yet. do we have now also instructions how to deploy a contract only with bytecode and aci? should be possible now, right?

Reply:
> is that still correct with the source as part of the object changes?

Nope, the changes are explained as they happen in history. I've fixed it by moving it below than putting `source` into options.